### ISSUE TITLE
All translatable fields of a form are empty strings

### ISSUE THREAD

**Issue by SilviaAmAm** [open-formulieren/open-forms#2339]
### Product versie / Product version

latest (cdc5b564e647080b082940a5ec3c9dcb15dd39d5)

### Omschrijf het probleem / Describe the bug

As I try to open a new form in the SDK, all translatable fields of the form are empty (form name, step names)
![Screenshot from 2022-11-10 16-09-01](https://user-images.githubusercontent.com/19154114/201131796-cb100c0f-43d6-43cd-9206-c8fff4167a4c.png)

The `LANGUAGE_CODE` is set to `nl`.

The `Content-language` header of the response for the GET to the form endpoint is nl

If I comment out this line of code: https://github.com/open-formulieren/open-forms/blob/master/src/openforms/forms/api/viewsets.py#L422 (and I delete the language cookie) then the strings are no longer empty.

### Stappen om te reproduceren / Steps to reproduce

Look at any form using the local SDK (33c6e82d1fed892abad07be6cda9d8149abd1899)

### Verwacht gedrag / Expected behavior

The translatable fields should not be turned into empty strings

### Device

Desktop

### OS

Linux

### Browser

Firefox

**Comment by stevenbal** [comment]
After some debugging, we noticed that there was a difference in behaviour when creating Forms via `localhost` vs `127.0.0.1`. The former had its locale set to English for some reason and the latter to Dutch. 

When creating a Form on `127.0.0.1`, the values for the translatable fields are all saved to `<fieldname>_en`, because of the locale and because the Form builder form does not support translatable fields yet. When loading the form using the SDK, with locale set to `nl`, only empty strings are shown, because the translatable fields do not have values for Dutch yet.

TL;DR: when creating Forms locally, we should ensure that the locale is set to Dutch for now, until the Form builder supports setting values for literals in multiple languages (https://github.com/open-formulieren/open-forms/issues/2314)
